internal/service/dynamodb/global_secondary_index.go

```diff
@@ -0,0 +1,934 @@
+				Required:   true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"range_key": schema.StringAttribute{
+				Optional: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"range_key_type": schema.StringAttribute{
+				CustomType: fwtypes.StringEnumType[ddbtypes.ScalarAttributeType](),
+				Optional:   true,
+				Computed:   true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"read_capacity": schema.Int64Attribute{
+				Optional: true,
+				Computed: true,
+			},
+			"table": schema.StringAttribute{
+				Required: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"write_capacity": schema.Int64Attribute{
+				Optional: true,
+				Computed: true,
+			},
+		},
+		Blocks: map[string]schema.Block{
+			"on_demand_throughput": schema.ListNestedBlock{
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"max_read_request_units": schema.Int64Attribute{
+							Optional: true,
+							Computed: true,
+						},
+						"max_write_request_units": schema.Int64Attribute{
+							Optional: true,
+							Computed: true,
+						},
+					},
+				},
+				CustomType: fwtypes.NewListNestedObjectTypeOf[onDemandThroughputModel](ctx),
+				Validators: []validator.List{
+					listvalidator.SizeAtMost(1),
+				},
+				PlanModifiers: []planmodifier.List{
+					listplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"warm_throughput": schema.ListNestedBlock{
+				CustomType: fwtypes.NewListNestedObjectTypeOf[warmThroughputModel](ctx),
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"read_units_per_second": schema.Int64Attribute{
+							Optional: true,
+							Computed: true,
+						},
+						"write_units_per_second": schema.Int64Attribute{
+							Optional: true,
+							Computed: true,
+						},
+					},
+				},
+				Validators: []validator.List{
+					listvalidator.SizeAtMost(1),
+				},
+				PlanModifiers: []planmodifier.List{
+					listplanmodifier.UseStateForUnknown(),
+				},
+			},
+		},
+		Version: 1,
+	}
+
+	if s.Blocks == nil {
+		s.Blocks = make(map[string]schema.Block)
+	}
+	s.Blocks[names.AttrTimeouts] = timeouts.Block(ctx, timeouts.Opts{
+		Create: true,
+		Update: true,
+		Delete: true,
+	})
+
+	response.Schema = s
+}
+
+func (r *resourceGlobalSecondaryIndex) Create(ctx context.Context, request resource.CreateRequest, response *resource.CreateResponse) {
+	var data resourceGlobalSecondaryIndexModel
+
+	response.Diagnostics.Append(request.Plan.Get(ctx, &data)...)
+
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	conn := r.Meta().DynamoDBClient(ctx)
+	if _, err := waitAllGSIActive(ctx, conn, data.Table.ValueString(), r.ReadTimeout(ctx, data.Timeouts)); err != nil {
```

Comment:
Even though `waitAllSGIActive` is technically a read call, it's part of the resource's Create operation, so we should use the `CreateTimeout` here


```suggestion
	if _, err := waitAllGSIActive(ctx, conn, data.Table.ValueString(), r.CreateTimeout(ctx, data.Timeouts)); err != nil {
```